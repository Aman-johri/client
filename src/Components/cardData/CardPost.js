import React from "react";
import { Link } from "react-router-dom";
import "./post.css";
import Tooltip from "@mui/material/Tooltip";
import Button from '@mui/material/Button';
import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogContent from '@mui/material/DialogContent';
import DialogContentText from '@mui/material/DialogContentText';
import DialogTitle from '@mui/material/DialogTitle';
import { deleteData } from "../../redux/actions/postActions";
import { connect } from 'react-redux';
import SinglePost from "../singlePost/SinglePost";
import PropTypes from "prop-types";
import EditPostInfo from "../editPostInfo/index";


class Post extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            open: false,
            postOpen: false,
            open2: false
        };
        this.handleClickOpen = this.handleClickOpen.bind(this);
        this.handleClose = this.handleClose.bind(this);
    }

    handleClickOpen2 = () => {
        this.setState({ open2: true });
    };

    handleClickOpen = () => {
        this.setState({ open: true });
    };

    handleClose = () => {
        this.setState({ open: false });
    };

    handleClickEdit = () => {
        this.setState({ postOpen: true , open2: false});
    };
    render() {
        return (
            <>
                <div className="post">
                    <img className="postImg"
                        src={this.props.post.img} />
                    <div className="postInfo">
                        <div className='PostEdit'>
                            <i className='singlePostIcon1 fa fa-edit' onClick={this.handleClickOpen2} id='edit'></i>
                            <i className='singlePostIcon2 fa fa-trash' id='delete' onClick={this.handleClickOpen}></i>
                        </div>
                        <div className="postCats">
                            <span className="postCat">{this.props.post.categories}</span>
                        </div>
                        <Tooltip title={<h3>{this.props.post.title}</h3>} placement="right">
                            <span className="postTitle">{this.props.post.title.slice(0, 15)}{this.props.post.title.length > 15 ? " ...." : ""}</span>
                        </Tooltip>
                        <span className="postDate">{new Date(this.props.post.createdAt).toDateString()}</span>
                        <button className='singlePostIcon3' onClick={this.handleClickEdit}>View Blog</button>
                    </div>
                    <Dialog
                        open={this.state.open}
                        onClose={this.handleClose}
                        aria-labelledby="alert-dialog-title"
                        aria-describedby="alert-dialog-description" >
                        <DialogTitle id="alert-dialog-title">
                            {"Are you sure you want to delete this post?"}
                        </DialogTitle>
                        <DialogContent>
                            <DialogContentText id="alert-dialog-description">
                                This action cannot be undone.
                            </DialogContentText>
                        </DialogContent>
                        <DialogActions>
                            <Button onClick={this.handleClose}>Cancel</Button>
                            <Button onClick={deleteData(this.props.post._id)}>
                                Continue
                            </Button>
                        </DialogActions>
                    </Dialog>
                </div>
                <SinglePost id={this.props.post._id} open={this.state.postOpen} img={this.props.post.img} post={this.props.post} />
                <EditPostInfo id={this.props.post._id} open={this.state.open2} img={this.props.post.img} post={this.props.post} />
            </>
        );
    }
}

// const mapDispatchToProps = dispatch => ({
//     deleteData: (id) => dispatch(deleteData(id))
// })

Post.propTypes = {
    deleteData: PropTypes.func.isRequired
}

Post.defaultProps = {
    deleteData: () => { },
}

// export default connect(null,mapDispatchToProps)(Post);
export default Post;