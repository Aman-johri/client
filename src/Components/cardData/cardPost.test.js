import { render } from "@testing-library/react";
import CardPost from "./CardPost";
import {shallow,configure} from 'enzyme';
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
configure({ adapter: new Adapter() });

describe("Testing CardPost component", () => {
    it("snapshot test",() => {
        const component = shallow(<CardPost post={{title:"Aman Johri", createdAt:"Sat Apr 02 2022", categories:"Tech"}} />);
        const tree = component.debug();
        expect(tree).toMatchSnapshot();
    });

    it("check createdAt date", () => {
        const component = shallow(<CardPost post={{title:"Aman Johri", createdAt:"Sat Apr 02 2022", categories:"Tech" ,_id:"166354879548566255" }} />);
        expect(component.find(".postDate").text()).toEqual("Sat Apr 02 2022");
    });

    it("check title",() => {
        const component = shallow(<CardPost post={{title:"Aman Johri", createdAt:"Sat Apr 02 2022", categories:"Tech" ,_id:"166354879548566255" }} />);
        expect(component.find(".postTitle").text()).toEqual("Aman Johri");
    })

    it("check categories",() => {
        const component = shallow(<CardPost post={{title:"Aman Johri", createdAt:"Sat Apr 02 2022", categories:"Tech" ,_id:"166354879548566255" }} />);
        expect(component.find(".postCat").text()).toEqual("Tech");
    })
})