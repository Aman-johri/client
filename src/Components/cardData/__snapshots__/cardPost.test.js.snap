// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CardPost component snapshot test 1`] = `
"<Fragment>
  <div className=\\"post\\">
    <img className=\\"postImg\\" src={[undefined]} />
    <div className=\\"postInfo\\">
      <div className=\\"PostEdit\\">
        <i className=\\"singlePostIcon1 fa fa-edit\\" onClick={[Function (anonymous)]} id=\\"edit\\" />
        <i className=\\"singlePostIcon2 fa fa-trash\\" id=\\"delete\\" onClick={[Function: bound ]} />
      </div>
      <div className=\\"postCats\\">
        <span className=\\"postCat\\">
          Tech
        </span>
      </div>
      <ForwardRef(Tooltip) title={{...}} placement=\\"right\\">
        <span className=\\"postTitle\\">
          Aman Johri
        </span>
      </ForwardRef(Tooltip)>
      <span className=\\"postDate\\">
        Sat Apr 02 2022
      </span>
      <button className=\\"singlePostIcon3\\" onClick={[Function (anonymous)]}>
        View Blog
      </button>
    </div>
    <ForwardRef(Dialog) open={false} onClose={[Function: bound ]} aria-labelledby=\\"alert-dialog-title\\" aria-describedby=\\"alert-dialog-description\\">
      <ForwardRef(DialogTitle) id=\\"alert-dialog-title\\">
        Are you sure you want to delete this post?
      </ForwardRef(DialogTitle)>
      <ForwardRef(DialogContent)>
        <ForwardRef(DialogContentText) id=\\"alert-dialog-description\\">
          This action cannot be undone.
        </ForwardRef(DialogContentText)>
      </ForwardRef(DialogContent)>
      <ForwardRef(DialogActions)>
        <ForwardRef(Button) onClick={[Function: bound ]}>
          Cancel
        </ForwardRef(Button)>
        <ForwardRef(Button) onClick={[Function (anonymous)]}>
          Continue
        </ForwardRef(Button)>
      </ForwardRef(DialogActions)>
    </ForwardRef(Dialog)>
  </div>
  <SinglePost id={[undefined]} open={false} img={[undefined]} post={{...}} deleteData={[Function: deleteData]} getData={[Function: getData]} updateData={[Function: updateData]} />
  <Connect(SinglePost) id={[undefined]} open={false} img={[undefined]} post={{...}} />
</Fragment>"
`;
